### ISSUE TITLE
[apex] ApexCRUDViolation incorrectly cleared for subsequent queries in method

### ISSUE THREAD

**Issue by yarhamjohn** [pmd/pmd#4818]
**Affects PMD Version:** 
7.0.0-rc3

**Rule:**
ApexCRUDViolation - https://pmd.github.io/pmd/pmd_rules_apex_security.html#apexcrudviolation

**Description:**
PMD correctly highlights all DML update statements in the method as missing permissions checks. However, when the first of these is fixed as per the docs, _all_ the other PMD errors in the file get cleared even though they should still be problems.

**Code Sample demonstrating the issue:**

Both update statements are highlighted by PMD as having problems (correct):
```java
public void myMethod()
{
    update new Account(Name = 'Account Name');
    update new Account(Name = 'Account Name 2');
}
```

The first update is no longer highlighted by PMD (correct), but nor is the second (incorrect).
```java
public void myMethod()
{
    if (Schema.sObjectType.Account.isUpdateable()) {
        update new Account(Name = 'Account Name');
    }
    
    update new Account(Name = 'Account Name 2');
}
```

**Expected outcome:**

PMD should report a violation at the second update line because there is no check that guarantees that Account can be updated in that code path, but doesn't. This is a false-negative.

**Running PMD through:** VSCode using https://marketplace.visualstudio.com/items?itemName=chuckjonas.apex-pmd (v0.6.2)
